### PR TITLE
Add the client side bits of kubectl export functionality

### PR DESCRIPTION
**Pull request by brendandburns**
@smarterclayton @bgrant0607 @janetkuo @kubernetes/kubectl 

Ref: https://github.com/kubernetes/kubernetes/issues/13038

### CODE REVIEW

**Comment by bgrant0607** [comment]
Assigned to @smarterclayton to choose a reviewer.

**Comment by smarterclayton** [comment]
@deads2k do you have bandwidth in the next day or two?

**Comment by deads2k** [comment]
> @deads2k do you have bandwidth in the next day or two?

Yes, I'll look through.

**Comment by deads2k** [comment]
I'd like to see a test in `test-cmd` that demonstrates the field stripping effects of `kubectl get --export`.

Otherwise, just one nit on the second commit.

**Comment by deads2k** [comment]
@brendandburns unit test failures are real.

**Comment by deads2k** [comment]
I'd still really like to see a test in `test-cmd`: https://github.com/kubernetes/kubernetes/pull/18101#issuecomment-161971276

**Comment by brendandburns** [comment]
@k8s-bot unit test this please

**Comment by brendandburns** [comment]
@deads2k friendly ping.

File: pkg/kubectl/resource/visitor.go [hunk 1]
```diff
@@ -103,8 +106,8 @@ func (i *Info) Visit(fn VisitorFunc) error {
 }
 
 // Get retrieves the object from the Namespace and Name fields
-func (i *Info) Get() error {
-	obj, err := NewHelper(i.Client, i.Mapping).Get(i.Namespace, i.Name)
+func (i *Info) Get() (err error) {
```
Comment (deads2k) [inline pkg/kubectl/resource/visitor.go]: Keep this as an un-named return value.  You're not using the feature and it often leads to unusual name shadowing.

**Comment by deads2k** [comment]
You might need to do another push.  I'm not seeing the `test-cmd.sh` update.

**Comment by deads2k** [comment]
Once there's a `test-cmd.sh` test case, feel free to tag, the code LGTM

File: pkg/kubectl/resource/visitor.go (hunk 1 above)
Comment (brendandburns) [inline pkg/kubectl/resource/visitor.go]: done.

**Comment by brendandburns** [comment]
@deads2k thanks, sorry, I pushed from my mac, test-cmd.sh only works on Linux, so the changes were there.  re-pushed.

--brendan

**Comment by deads2k** [comment]
needs a squash, then LGTM

**Comment by brendandburns** [comment]
@deads2k are you a git guru, I can't seem to figure out how to squash since the top commit is a merge...

**Comment by deads2k** [comment]
> @deads2k are you a git guru, I can't seem to figure out how to squash since the top commit is a merge...

Sorry, I'm not a git guru, but something seems wrong.  That "merge" commit is huge and seems to modify a bunch of files not in the diff for the pull request.  Any chance its just a mistake and if you drop it your branch still has all the expected changes and works?

**Comment by brendandburns** [comment]
@deads2k ok, I'll try to re-create this from scratch somehow...

**Comment by brendandburns** [comment]
@deads2k ok, I copied the test-cmd.sh changes into my clean-ish client on my mac and re-pushed.  Hopefully that looks better.

--brendan

**Comment by kargakis** [comment]
```
Summarizing 1 Failure:

[Fail] Services [It] should be able to up and down services 
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/service.go:244

Ran 116 of 208 Specs in 477.298 seconds
FAIL! -- 115 Passed | 1 Failed | 2 Pending | 90 Skipped 
```
Probably a flake

**Comment by wojtek-t** [comment]
@k8s-bot e2e test this please

**Comment by spxtr** [comment]
@k8s-bot unit test this please

#18960

**Pull request merged**